### PR TITLE
[INLONG-8021][Manager] Periodically delete sources with inconsistent states

### LINKED ISSUE #8021: [Improve][Manager] Periodically delete sources with inconsistent states
### Description

Periodically delete sources with inconsistent states.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #8021 

### Motivation

Periodically delete sources with inconsistent states.

### Modifications

Periodically delete sources with inconsistent states.

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 1]
```diff
@@ -663,4 +670,22 @@ public void run() {
...
+     * update task which in an incorrect state
+     */
+    private class TaskCleanRunnable implements Runnable {
+
+        @Override
+        public void run() {
+            while (true) {
+                try {
+                    sourceMapper.updateStatusByDeleted();
+                    Thread.sleep(60 * 10000);
```
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java]: Fixed

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 2]
```diff
@@ -138,11 +140,16 @@ public class AgentServiceImpl implements AgentService {
...
     private void startHeartbeatTask() {
-        if (enabled) {
+        if (updateTaskTimeoutEnabled) {
             UpdateTaskRunnable taskRunnable = new UpdateTaskRunnable();
             this.executorService.execute(taskRunnable);
             LOGGER.info("update task status started successfully");
         }
+        if (sourceCleanEnabled) {
+            TaskCleanRunnable taskCleanRunnable = new TaskCleanRunnable();
+            this.executorService.execute(taskCleanRunnable);
```
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java]: done

**Review by vernedeng — APPROVED** [review]
LGTM

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 3]
```diff
@@ -138,11 +144,30 @@ public class AgentServiceImpl implements AgentService {
...
+        if (updateTaskTimeoutEnabled) {
+            ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+            executor.scheduleWithFixedDelay(() -> {
+                try {
+                    sourceMapper.updateStatusToTimeout(beforeSeconds);
+                    LOGGER.info("update task status successfully");
+                } catch (Throwable t) {
+                    LOGGER.error("update task status error", t);
+                }
+            }, 0, beforeSeconds, TimeUnit.SECONDS);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:156]: 1. Suggest add a new interval params, not reuse the `beforeSeconds`.
2. Please add a name for the thread pool, such as: 
    ```java
    ThreadFactory factory = new ThreadFactoryBuilder()
            .setNameFormat("scheduled-source-timeout-%d")
            .setDaemon(true)
            .build();
    ScheduledExecutorService executor  = Executors.newSingleThreadScheduledExecutor(factory);
    ```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 4]
```diff
@@ -138,11 +144,30 @@ public class AgentServiceImpl implements AgentService {
...
             LOGGER.info("update task status started successfully");
         }
+        if (sourceCleanEnabled) {
+            ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:160]: Name to "scheduled-source-deleted-%d".

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 3 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 4 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java]: done

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**